### ISSUE TITLE
Adding a "collection_id" field to an Assignment?

### ISSUE THREAD

**Issue by Z4P0** [hhroc/yellr#178]
Every time we publish an assignment, we also create a collection. That collection is tied to the assignment. With out current setup I can't think of a way to know which collection belongs to which assignment. We have a list of collections and a list of assignments, but no way to check which goes with which.

Maybe we could have an "admin/set_collection_id.json" endpoint? Where we would pass the assignment_id and the collection_id.

For the demo on Saturday we can assume the assignment_id and collection_id will always be the same. ex. Create first assignment (assignment_id=1), which creates the first collection (colllection_id=1)

**Comment by thequbit** [comment]
Herm ... I wasn't thinking that a collection would be created for each
assignment every time, so I didn't think to link them.  I think that
assumption will work okay for Saturday, but could get us in trouble in the
future.

Perhaps a check box for "create a collection with this assignment?".  And
then we can make a table that links Assignments to Collections (that way
you could have multiple Assignments assigned to a single Collection).  Does
that make sense?

On Thu, Sep 11, 2014 at 2:39 PM, luis notifications@github.com wrote:

> Every time we publish an assignment, we also create a collection. That
> collection is tied to the assignment. With out current setup I can't think
> of a way to know which collection belongs to which assignment. We have a
> list of collections and a list of assignments, but no way to check which
> goes with which.
> 
> Maybe we could have an "admin/set_collection_id.json" endpoint? Where we
> would pass the assignment_id and the collection_id.
> 
> For the demo on Saturday we can assume the assignment_id and collection_id
> will always be the same. ex. Create first assignment (assignment_id=1),
> which creates the first collection (colllection_id=1)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hhroc/yellr/issues/178.

## 

_Tim Duffy_
http://timduffy.me/
(585)-210-8353
@arbiterofbits

**Comment by Z4P0** [comment]
Yeah, I was actually thinking something along those lines as well, like
having the checkbox, but "reversed". The default action should create a new
collection, but if the user unchecks the box it will not create a new one,
and allow them to choose which existing collection to use instead

On Thu, Sep 11, 2014 at 2:50 PM, Timothy Duffy notifications@github.com
wrote:

> Herm ... I wasn't thinking that a collection would be created for each
> assignment every time, so I didn't think to link them. I think that
> assumption will work okay for Saturday, but could get us in trouble in the
> future.
> 
> Perhaps a check box for "create a collection with this assignment?". And
> then we can make a table that links Assignments to Collections (that way
> you could have multiple Assignments assigned to a single Collection). Does
> that make sense?
> 
> On Thu, Sep 11, 2014 at 2:39 PM, luis notifications@github.com wrote:
> 
> > Every time we publish an assignment, we also create a collection. That
> > collection is tied to the assignment. With out current setup I can't
> > think
> > of a way to know which collection belongs to which assignment. We have a
> > list of collections and a list of assignments, but no way to check which
> > goes with which.
> > 
> > Maybe we could have an "admin/set_collection_id.json" endpoint? Where we
> > would pass the assignment_id and the collection_id.
> > 
> > For the demo on Saturday we can assume the assignment_id and
> > collection_id
> > will always be the same. ex. Create first assignment (assignment_id=1),
> > which creates the first collection (colllection_id=1)
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/hhroc/yellr/issues/178.
> 
> ## 
> 
> _Tim Duffy_
> http://timduffy.me/
> (585)-210-8353
> @arbiterofbits
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hhroc/yellr/issues/178#issuecomment-55309896.

## 

Luis Rosario
@zapo_tec
http://www.luisrosar.io/
